cmd/unused-exporter/exporter_test.go

```diff
@@ -0,0 +1,226 @@
+package main
+
+import (
+	"context"
+	"log/slog"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/grafana/unused"
+	"github.com/grafana/unused/aws"
+	"github.com/grafana/unused/azure"
+	"github.com/grafana/unused/gcp"
+)
+
+type MockProvider struct {
+	name string
+}
+
+type MockDisk struct {
+	name      string
+	sizeGB    int
+	createdAt time.Time
+	meta      map[string]string
+}
+
+func (p *MockProvider) Name() string {
+	return p.name
+}
+
+func (p *MockProvider) Delete(context context.Context, d unused.Disk) error {
+	return nil
+}
+
+func (p *MockProvider) ID() string {
+	return ""
+}
+
+func (p *MockProvider) ListUnusedDisks(ctx context.Context) (unused.Disks, error) {
+	return nil, nil
+}
+
+func (p *MockProvider) Meta() unused.Meta {
+	return unused.Meta{}
+}
+
+func (d *MockDisk) Name() string {
+	return d.name
+}
+
+func (d *MockDisk) Meta() unused.Meta {
+	return d.meta
+}
+
+func (d *MockDisk) CreatedAt() time.Time {
+	return d.createdAt
+}
+
+func (d *MockDisk) ID() string {
+	return ""
+}
+
+func (d *MockDisk) Provider() unused.Provider {
+	return nil
+}
+
+func (d *MockDisk) SizeGB() int {
+	return d.sizeGB
+}
+
+func (d *MockDisk) SizeBytes() float64 {
+	return 0
+}
+
+func (d *MockDisk) LastUsedAt() time.Time {
+	return time.Time{}
+}
+
+func (d *MockDisk) DiskType() unused.DiskType {
+	return ""
+}
+
+func TestGetRegionFromZone(t *testing.T) {
+	type testCase struct {
+		name     string
+		provider string
+		zone     string
+		expected string
+	}
+
+	testCases := []testCase{
+		{
+			name:     "Azure",
+			provider: azure.ProviderName,
+			zone:     "eastus1",
```

Comment:
@paulajulve is it the case that zone == region in Azure? (as per previous code, also kept in this PR)

Reply:
cool, tnx